apps/back/src/Entity/User.php

```diff
@@ -3,32 +3,51 @@
 namespace App\Entity;
 
 use App\Repository\UserRepository;
+use DateTime;
 use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
 use Symfony\Component\Security\Core\User\UserInterface;
 use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
 
 #[ORM\Entity(repositoryClass: UserRepository::class)]
 #[UniqueEntity(fields: ['email'], message: 'There is already an account with this email')]
 class User implements UserInterface, PasswordAuthenticatedUserInterface
 {
-
     #[ORM\Id]
     #[ORM\GeneratedValue]
     #[ORM\Column(type: 'integer')]
     private int $id;
-
+    #[Assert\Email(
+        message: 'The email {{ value }} is not a valid email.',
+    )]
     #[ORM\Column(type: 'string', length: 180, unique: true)]
     private ?string $email;
+    #[ORM\Column(type: 'string', length: 180, unique: true)]
+    private ?string $username;
 
     #[ORM\Column(type: 'json')]
     private array $roles = [];
-
+    
+    // TODO uncomment for production
+//    #[Assert\PasswordStrength]
     #[ORM\Column(type: 'string')]
     private string $password;
 
     #[ORM\Column(type: 'boolean')]
-    private $isVerified = false;
+    private bool $isVerified = false;
+
+    #[ORM\Column(type: "datetime")]
+    private DateTime $registerAt;
+    #[ORM\Column(type: 'integer')]
+    private int $videoViewed;
+
+    public function __construct()
+    {
+        $this->registerAt = new DateTime();
```

Comment:
Attention à DateTime !! C'est un objet non immutable, donc tu peux changer sa valeur en faisant `$user->getRegisterAt()->modify('mon cul sur la commode')`

Reply:
Attention ça ne fonctionne pas !! 
`Warning: DateTime::modify(): Failed to parse time string (mon cul sur la commode) at position 4 (c): The timezone could not be found in the database`